lib/src/typewriter.dart

```diff
@@ -61,8 +66,18 @@ class TypewriterAnimatedText extends AnimatedText {
   /// Widget showing partial text
   @override
   Widget animatedBuilder(BuildContext context, Widget child) {
+    /// Output of CurveTween is in the range [0, 1] for majority of the curves.
+    /// It is converted to [0, textCharacters.length].
     final textLen = textCharacters.length;
-    final typewriterValue = _typewriterText.value;
+    var typewriterValue =
+        (_typewriterText.value * textCharacters.length).round();
+
+    if (typewriterValue < 0) {
+      typewriterValue = 0;
+    }
+    if (typewriterValue >= textCharacters.length) {
```

Comment:
Line 86 compensates for a value exceeding the length, so this _if_ is unnecessary.

Reply:
I guess the example I gave above is applicable here as well.